### PR TITLE
Calculate solar azimuth

### PR DESCRIPTION
**Pull request by jareyes**
Because people want to know where the sun is.

### CODE REVIEW

File: lib/solar.js [hunk 1]
```diff
@@ -147,6 +150,26 @@ function sin_elevation(ms, lat, lon) {
...
+// https://midcdmz.nrel.gov/spa/spa.h
+function azimuth(ms, lat, lon) {
+  const t = julian.to(ms);
+  const sin_lat = Math.sin(lat * RAD);
+  const cos_lat = Math.cos(lat * RAD);
+  const hour_angle = sidereal.local(t, lon * RAD) - _right_ascension(t);
+  const sin_hour_angle = Math.sin(hour_angle);
+  const cos_hour_angle = Math.cos(hour_angle);
+  const dec = _declination(t);
+  const tan_dec = Math.tan(dec);
```
Comment (ironwallaby) [inline lib/solar.js]: This is the only real thing I dislike. It's fine for a first-pass, but once you got it working, did you test the alternative way?

```javascript
const sin_dec = _sin_declination(t);
const tan_dec = sin_dec / Math.sqrt(1 - sin_dec * sin_dec);
```

File: lib/solar.js [hunk 2]
```diff
@@ -147,6 +150,26 @@ function sin_elevation(ms, lat, lon) {
...
+
+  return Math.PI + Math.atan2(
+    sin_hour_angle,
+    cos_hour_angle * sin_lat - tan_dec * cos_lat
+  );
+}
+
+
 // Solar position classes. Helpful if you need to cache results!
 class Location {
```
Comment (ironwallaby) [inline lib/solar.js]: I would love it if the `Location` class included azimuth. That's probably work for another day, though.

File: lib/solar.js (hunk 1 above)
Comment (ironwallaby) [reply lib/solar.js]: Also you should print this Wikipedia page out and hang it by your desk: https://en.wikipedia.org/wiki/List_of_trigonometric_identities

Comment (jareyes) [reply lib/solar.js]: That passed the tests, too!

**Pull request merged**